app/src/main/java/com/esri/arcgisruntime/opensourceapps/datacollection/views/popup/PopupView.kt

```diff
@@ -71,28 +128,164 @@ class PopupView : FrameLayout {
+                val featureTable: FeatureTable = feature.featureTable
+                if (featureTable is ServiceFeatureTable) {
+                    val applyEditsFuture: ListenableFuture<List<FeatureEditResult>> =
+                        featureTable.applyEditsAsync()
+                    applyEditsFuture.addDoneListener {
+                        try {
+                            val featureEditResults: List<FeatureEditResult> = applyEditsFuture.get()
+                            // Check for errors in FeatureEditResults
+                            if (featureEditResults.any { result -> result.hasCompletedWithErrors() }) {
+                                showSnackbarMessage("There was a problem saving the feature")
+                            } else {
+                                showSnackbarMessage("Feature saved successfully")
+                            }
+
+                        } catch (ie: InterruptedException) {
+                            Logger.e(
+                                "Exception encountered when saving popup, " + Util.getExceptionMessageAndCause(
+                                    ie
+                                )
+                            )
+                        } catch (ee: ExecutionException) {
+                            Logger.e(
+                                "Exception encountered when saving popup, " + Util.getExceptionMessageAndCause(
+                                    ee
+                                )
+                            )
+                        }
+                    }
+                }
+            } catch (ie: InterruptedException) {
+                Logger.e(
+                    "Exception encountered when saving popup, " + Util.getExceptionMessageAndCause(
+                        ie
+                    )
                 )
-            )
+            } catch (ee: ExecutionException) {
+                Logger.e(
+                    "Exception encountered when saving popup, " + Util.getExceptionMessageAndCause(
+                        ee
+                    )
+                )
+            }
+        }
+    }
+
+    /**
+     * Updates the given popup field value with the appropriately cast string value.
+     *
+     * @param manager - PopupManager
+     * @param field   - PopupField
+     * @param value   - String
+     * @return - ArcGISRuntimeException if there's a problem updating the field value,
+     * otherwise null.
+     */
+    private fun updateValue(
+        manager: PopupManager,
+        field: PopupField?,
+        value: String?
+    ): ArcGISRuntimeException? {
+        var error: ArcGISRuntimeException? = null
+
+        when (manager.getFieldType(field)) {
+            Field.Type.SHORT -> error = if (value == null || value.trim { it <= ' ' }.isEmpty()) {
+                manager.updateValue(null, field)
+            } else {
+                manager.updateValue(value.toString().toShort(), field)
```

Comment:
I don't think there's need to call `toString()` again since value is already a String, applies to all others below 
```suggestion
                manager.updateValue(value.toShort(), field)
```

Reply:
I have removed this code. No longer applicable.